pkg/controller/ecs/service/setup.go

```diff
@@ -80,6 +80,11 @@ func postObserve(_ context.Context, cr *svcapitypes.Service, resp *svcsdk.Descri
 		return obs, err
 	}
 
+	if resp.Services[0].DesiredCount != resp.Services[0].RunningCount {
+		cr.SetConditions(xpv1.Creating())
+		return obs, nil
+	}
+
```

Comment:
Sorry, for being picky here, but I am not sure if we would want to return here already. What if the state is `INACTIVE` and the number of tasks does not match? We would never run into `obs.ResourceExists = false` in L96 probably preventing deletion of the resource?

I think best would be to put this `if` clause into the `case "ACTIVE"` block below since that is the only time were it is relevant to check if the number of desired tasks is running.

Reply:
Hello @MisterMX , good catch.

Updated the PR accordingly.